torch_geometric/loader/link_neighbor_loader.py

```diff
@@ -11,7 +11,42 @@
 
 
 class LinkNeighborSampler(NeighborSampler):
+    def __init__(self, data: Union[Data, HeteroData],
+                 num_neighbors: NumNeighbors, replace: bool = False,
+                 directed: bool = True, input_type: Optional[Any] = None,
+                 share_memory: bool = False, neg_size: int = 0):
```

Comment:
I would have preferred to not have `neg_size` here, but instead again use `neg_sampling_ratio`, but to make the batch size not change, this would have meant modifying the ratio which I thought was confusing.

Reply:
I think we should just get the batch size directly in the sampler. This should be doable via `edge_label_index.size(1)`.

Reply:
So my thinking was, if we provide `neg_sampling_ratio` but don't adjust the batch size, then we'd need to either increase the size of the batch or throw away some of the batch: Throwing away seemed like something people might not want to do.

But, happy to adapt to either of these. What do you think?

Reply:
Ah, I see. I think there is no reason to modify `batch_size` according to `neg_sampling_ratio` since I personally would have expected that `batch_size` only refers to positive examples. I get your point though. We can also modify `batch_size` in `__init__` and pass `neg_sampling_ratio` down the line to `LinkNeighborSampler`.

Reply:
Yeah thats a little cleaner.. although I'd need to probably modify `neg_sampling_ratio` which is a bit confusing.

Example
- `batch_size=5`
- `neg_sampling_ratio=0.5`
- `new_batch_size=3`
- `new_neg_sampling_ratio= 2/3`

the new `new_neg_sampling_ratio` is needed because of the rounding...

But if you think just adding to the batch is natural for users, I like this better, I'll change to that :-)

Reply:
If we are operating on fixed `neg_size`, then this will also be a problem for the last batch, so there is always some kind of trade-off we make :)

Reply:
done

Reply:
> If we are operating on fixed neg_size, then this will also be a problem for the last batch, so there is always some kind of trade-off we make :)

Yep good point. I think whatever is most predictable to users is probably the best.